### PR TITLE
Close the YAML-ingress gaps RT-5/RT-6 did not reach (RT-7, RT-8, RT-9, MODEL-1)

### PR DESCRIPTION
**Pull request by anentropic**
Stacked on #217. **Contains a breaking change (RT-7) that I'd particularly like your call on — see the end.**

#209 enforced the YAML → GET_DDL round-trip contract for identifier **slots**. This closes the fields that are not slots, where a YAML-valid definition still rendered DDL that replays to a *different* model or none at all.

## RT-8 (HIGH) — a `--` in a YAML member expression silently destroyed sibling members

`member_expr` lacked the `blank_sql_comments` rejection `slot_common` has, and `column_roundtrips_verbatim` has no comment awareness (as `model.rs` itself documents). So YAML with `expr: 'o.a -- x'` passed the gate, rendered `o.d1 AS o.a -- x,\n o.d2 AS o.b`, and on replay the front door blanked from `--` through the member-separating comma: **two dimensions became one**, with d2's definition swallowed into d1's expression. Parse succeeded. No error anywhere.

## MODEL-1 — YAML import bypassed every cross-reference validation

Those checks lived only in `parse_keyword_body`, so six shapes passed YAML validation and then rendered DDL this extension's own parser rejects: ghost `NON ADDITIVE BY` dimension, ghost window inner metric, ghost materialization dimension/metric, duplicate materialization names, and a materialization with no members. Plus a **model-changing** one: a hostile `frame_clause` (`'ROWS … ) , junk AS ('`) replayed into **three metrics**, injecting one out of a string.

Fixed by hoisting the checks into `body_parser/cross_refs.rs`, called from **both** `parse_keyword_body` and `validate_ddl_representable` — shared implementation, not a second source of truth. Messages are byte-identical, so DDL-path diagnostics are unchanged.

For the window payload the fix verifies by **round trip** rather than a hazard list: render the spec, re-parse it with the real OVER parser, require equality. Complete by construction, zero false positives, and it subsumes the frame-clause injection *and* a partition dimension named `rows` silently becoming a frame clause — which no enumeration I'd have written would have caught.

## RT-9 — a derived metric named `private`/`public` could not be replayed

Rendered bare; the parser peels the token as the access modifier. Now quoted on emission when entry-initial (not unconditionally — that would store `"private"` where the definition said `private`, replacing one drift with another). `PUBLIC` has the identical peel and is handled; the review only named `PRIVATE`.

## ⚠️ RT-7 — `output_type` is now **refused** at YAML import

This is a documented, released field, so please sanity-check the reasoning. It was settable only from YAML and had no DDL emit site, so `GET_DDL` dropped it and a restored view silently lost its `CAST`.

The obvious alternative — render the cast into the emitted expression — was **measured, not assumed**, and is wrong twice over: member references splice a member's *raw* `expr` into referrers and never apply the cast, so a fact with `output_type: DOUBLE` emits `SUM((o.qty))`, not `SUM((CAST(o.qty AS DOUBLE)))` (pinned by a new test). Moving the cast into `expr` would silently change what every referring member computes — **and** the replayed definition would still carry `output_type: None`, leaving the contract false.

What tipped it: the field's only semantic effect is reproducible by writing `CAST(expr AS T)` into the expression, which round-trips correctly and applies everywhere the member is used. So no capability is lost, only a spelling whose restore path was broken. What *is* lost is the sole route to a populated `data_type` column; TECH-DEBT #51 is widened to record that read-side probing is now the only fix for it, and four doc pages are corrected.

**If you'd rather keep the field, this is the one commit to drop** — the rest of the PR stands without it, though `validate_ddl_representable`'s contract stays false for that field.

## Oracle strengthening

The existing oracles structurally could not see RT-7 or RT-8 (`roundtrip_proptest` parses via `parse_keyword_body` directly, never through front-door comment blanking; nothing diffed pre-render against replayed). `yaml_proptest` gains a YAML → render → **front-door** parse → model-equality oracle, watched red three separate ways.

Corrections to TECH-DEBT #58 and #60, whose contract claims were false when written.

---
_Generated by [Claude Code](https://claude.ai/code/session_016PT3CfUdQoRfc2SnvdzrbB)_

### CODE REVIEW

**Comment by anentropic** [comment]
Agreed — fixed. The rejection now prints both the stored spec and the re-parsed one.

The reasoning matches yours: the cases this fires on are subtle by construction — a `frame_clause` that escapes its parentheses, or a partition dimension named `rows` that reads back *as* a frame clause — and those are exactly the ones where "reads back as a different specification" leaves the author diffing two values they cannot see. `{:?}` on both is verbose, but it's the only thing that makes a whitespace or keyword-boundary case diagnosable from the error alone, which is the whole point of validating by round-trip rather than by a hazard list.

Model tests green (125 passed).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/model.rs [hunk 1]
```diff
@@ -764,20 +812,123 @@ fn member_expr(kind: &str, name: &str, expr: &str) -> Result<(), String> {
...
+        )
+    })?;
+    if replayed.as_ref() == Some(ws) {
+        Ok(())
+    } else {
+        Err(format!(
+            "metric '{metric_name}' has a window specification that does not \
+             round-trip: it renders as '{rendered}', which reads back as a \
+             DIFFERENT window specification"
+        ))
```
[the remark this answers is not in the document]
Comment (anentropic) [reply src/model.rs:930]: Agreed, and for exactly the reason you give — these rejections are subtle by nature (a `frame_clause` that escapes its parentheses, a partition dimension named `rows` that reads back as a frame), so "reads back as a DIFFERENT window specification" left the author diffing two invisible values.

Fixed in commit `d32a0f4`: the message now carries both, `stored:` and `re-parsed:`, each `{:?}`. Verbose, but it's the only thing that makes a whitespace or keyword-boundary case diagnosable from the error alone.

---
_Generated by [Claude Code](https://claude.ai/code)_
